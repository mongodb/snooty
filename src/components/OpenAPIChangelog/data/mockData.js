export const mockChangelog = [
  {
    date: '2024-02-01',
    paths: [
      {
        path: '/api/atlas/v2/groups/{groupId}/alertConfigs/{alertConfigId}',
        httpMethod: 'PUT',
        operationId: 'updateAlertConfiguration',
        tag: 'Alert Configurations',
        versions: [
          {
            version: '2023-01-01',
            stabilityLevel: 'stable',
            changeType: 'removed',
            changes: [
              {
                change: 'resource version 2023-01-01 was removed, latest available resource version is 2023-02-01',
                changeCode: 'resource-version-removed',
                backwardCompatible: true,
              },
            ],
          },
        ],
      },
    ],
  },
  {
    date: '2023-02-01',
    paths: [
      {
        path: '/api/atlas/v2/groups/byName/{groupName}',
        httpMethod: 'GET',
        operationId: 'getProjectByName',
        tag: 'Projects',
        versions: [
          {
            version: '2023-02-01',
            stabilityLevel: 'stable',
            changeType: 'release',
            changes: [
              {
                change: 'resource version added.',
                changeCode: 'resource-version-added',
                backwardCompatible: false,
              },
            ],
          },
        ],
      },
      {
        path: '/api/atlas/v2/groups/{groupId}/alertConfigs/{alertConfigId}',
        httpMethod: 'PUT',
        operationId: 'updateAlertConfiguration',
        tag: 'Alert Configurations',
        versions: [
          {
            version: '2023-02-01',
            stabilityLevel: 'stable',
            changeType: 'release',
            changes: [
              {
                change: 'resource version added.',
                changeCode: 'resource-version-added',
                backwardCompatible: false,
              },
              {
                change:
                  "for the 'query' request parameter 'clusterName', default value was changed from 'default' to 'cluster0'",
                changeCode: 'request-parameter-default-value-changed',
                backwardCompatible: false,
              },
              {
                change: "added the optional property 'links' in the response with the '200' status.",
                changeCode: 'response-optional-property-added',
                backwardCompatible: true,
              },
            ],
          },
          {
            version: '2023-01-01',
            stabilityLevel: 'stable',
            changeType: 'deprecate',
            sunsetDate: '2024-02-01',
            changes: [
              {
                change: 'deprecated by resource version 2023-02-01 and marked for removal for 2024-02-01.',
                changeCode: 'resource-version-deprecated',
                backwardCompatible: false,
              },
            ],
          },
        ],
      },
      {
        path: '/api/atlas/v2/groups/{groupId}/accessList',
        httpMethod: 'POST',
        operationId: 'createProjectIpAccessList',
        tag: 'Project IP Access List',
        versions: [
          {
            version: '2023-01-01',
            stabilityLevel: 'stable',
            changeType: 'update',
            changes: [
              {
                change: "added the optional property 'name' in the response with the '200' status.",
                changeCode: 'response-optional-property-added',
                backwardCompatible: true,
              },
              {
                change: "added the optional property 'comment' in the response with the '200' status.",
                changeCode: 'response-optional-property-added',
                backwardCompatible: true,
              },
            ],
          },
        ],
      },
    ],
  },
  {
    date: '2023-02-22',
    paths: [
      {
        path: '/api/atlas/v2/groups/byName/{groupName}',
        httpMethod: 'GET',
        operationId: 'getProjectByName',
        tag: 'Projects',
        versions: [
          {
            version: '2023-02-01',
            stabilityLevel: 'stable',
            changeType: 'update',
            changes: [
              {
                change:
                  "added the optional property 'withDefaultAlertsSettings' in the response with the '200' status.",
                changeCode: 'response-optional-property-added',
                backwardCompatible: true,
              },
            ],
          },
        ],
      },
    ],
  },
];

export const mockDiff = [
  {
    path: '/api/atlas/v2/federationSettings/{federationSettingsId}/connectedOrgConfigs',
    httpMethod: 'GET',
    operationId: 'listConnectedOrgConfigs',
    tag: 'Federated Authentication',
    changeType: 'update',
    changes: [
      {
        change: "removed the non-success response with the status '400'.",
        changeCode: 'response-non-success-status-removed',
        backwardCompatible: false,
      },
      {
        change: "added the new optional 'query' request parameter 'envelope'.",
        changeCode: 'new-optional-request-parameter',
        backwardCompatible: true,
      },
    ],
  },
  {
    path: '/api/atlas/v2/groups/{groupId}/accessList',
    httpMethod: 'POST',
    operationId: 'createProjectIpAccessList',
    tag: 'Project IP Access List',
    changeType: 'removed',
    changes: [
      {
        change: "removed the required property 'name' from the response with the '200' status.",
        changeCode: 'response-required-property-removed',
        backwardCompatible: false,
      },
    ],
  },
];

export const mockIndex = {
  runDate: '2023-04-03',
  runId: '20230403105948111',
  specRevision: '422ab47f864909549362a7b39688404c82e2540b',
  specRevisionShort: '422ab47f864',
  versions: ['2023-01-01', '2023-02-01', '2023-08-11~preview'],
};
