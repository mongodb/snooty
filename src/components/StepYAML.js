import React from 'react';
import PropTypes from 'prop-types';
import ComponentFactory from './ComponentFactory';

const StepYAML = ({ nodeData: { children }, stepNum, ...rest }) => (
  <div className="sequence-block">
    <div className="bullet-block">
      <div className="sequence-step">{stepNum}</div>
    </div>
    <div className="section">
      {children.map((child, index) => (
        <ComponentFactory {...rest} nodeData={child} key={index} />
      ))}
    </div>
  </div>
);

StepYAML.propTypes = {
  nodeData: PropTypes.shape({
    children: PropTypes.array.isRequired,
  }).isRequired,
  stepNum: PropTypes.number,
};

StepYAML.defaultProps = {
  stepNum: 1,
};

export default StepYAML;
