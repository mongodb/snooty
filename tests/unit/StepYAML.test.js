import React from 'react';
import { shallow } from 'enzyme';
import StepYAML from '../../src/components/StepYAML';

// data for this component
import mockData from './data/Step.test.json';

it('renders correctly', () => {
  const tree = shallow(<StepYAML nodeData={mockData} />);
  expect(tree).toMatchSnapshot();
});
