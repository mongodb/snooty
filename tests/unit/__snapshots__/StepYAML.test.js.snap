// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="sequence-block"
>
  <div
    className="bullet-block"
  >
    <div
      className="sequence-step"
    >
      1
    </div>
  </div>
  <div
    className="section"
  >
    <ComponentFactory
      key="0"
      nodeData={
        Object {
          "children": Array [
            Object {
              "children": Array [
                Object {
                  "position": Object {
                    "start": Object {
                      "line": 0,
                    },
                  },
                  "type": "text",
                  "value": "Connect to your MongoDB instance",
                },
              ],
              "position": Object {
                "start": Object {
                  "line": 0,
                },
              },
              "type": "heading",
            },
            Object {
              "argument": Array [
                Object {
                  "position": Object {
                    "start": Object {
                      "line": 0,
                    },
                  },
                  "type": "text",
                  "value": "/includes/drivers_whitelist.rst",
                },
              ],
              "children": Array [
                Object {
                  "children": Array [],
                  "position": Object {
                    "start": Object {
                      "line": 0,
                    },
                  },
                  "type": "FixedTextElement",
                },
              ],
              "name": "include",
              "position": Object {
                "start": Object {
                  "line": 0,
                },
              },
              "type": "directive",
            },
            Object {
              "argument": Array [
                Object {
                  "position": Object {
                    "start": Object {
                      "line": 2,
                    },
                  },
                  "type": "text",
                  "value": "/includes/drivers_connect.rst",
                },
              ],
              "children": Array [
                Object {
                  "children": Array [],
                  "position": Object {
                    "start": Object {
                      "line": 2,
                    },
                  },
                  "type": "FixedTextElement",
                },
              ],
              "name": "include",
              "position": Object {
                "start": Object {
                  "line": 2,
                },
              },
              "type": "directive",
            },
          ],
          "position": Object {
            "start": Object {
              "line": 0,
            },
          },
          "type": "section",
        }
      }
    />
  </div>
</div>
`;
